classes/Currency.php

```diff
@@ -423,31 +438,19 @@ public static function getCurrenciesByIdShop($idShop = 0)
      */
     protected static function addCldrDatasToCurrency($currencies, $isObject = false)
     {
-        $cldr = new Repository(Context::getContext()->language);
-
         if (is_array($currencies)) {
             foreach ($currencies as $k => $c) {
-                if ($isObject) {
-                    $currencies[$k] = Currency::getCurrencyInstance($c['id_currency']);
-                } else {
-                    $currency = $cldr->getCurrency($c['iso_code']);
-
-                    $currencies[$k]['name'] = ucfirst($currency['name']);
-                    $currencies[$k]['iso_code_num'] = $currency['iso_code'];
-                    $currencies[$k]['sign'] = $currency['symbol'];
-                    $currencies[$k]['format'] = '';
+                $currencies[$k] = Currency::getCurrencyInstance($c['id_currency']);
```

Comment:
maybe you should use `static` instead of `Currency` here?

Reply:
Both will have the same result.

Reply:
yes except for override no?

Reply:
Also with override.

With the PrestaShop autoloader, `Currency` will match an override if it exists, or the core class by default.

Reply:
Alright then :)